include/maths/CBoostedTree.h

```diff
@@ -391,6 +399,8 @@ class MATHS_EXPORT CBoostedTree final : public CDataFrameRegressionModel {
     //! Get the column containing the model's prediction for the dependent variable.
     std::size_t columnHoldingPrediction(std::size_t numberColumns) const override;
 
+    const TOptionalSizeVec& columnsHoldingShapValues() const override;
```

Comment:
How about just returning an empty vector if there aren't anything rather than making this optional? Seems cleaner.

Reply:
Indeed. However, this now becomes irrelevant as I let the result writer to pick top SHAP values for every row.